src/main/java/com/atherys/towns/util/Rectangle.java

```diff
@@ -0,0 +1,24 @@
+package com.atherys.towns.util;
+
+import com.flowpowered.math.vector.Vector2i;
+
+public interface Rectangle {
+    public Vector2i getTopLeftCorner();
+    public Vector2i getBottomRightCorner();
+
+    public default int maxX() {
+        return getBottomRightCorner().getX();
+    }
+
+    public default int minX() {
+        return getTopLeftCorner().getX();
+    }
+
+    public default int maxY() {
+        return getTopLeftCorner().getY();
+    }
+
+    public default int minY() {
```

Comment:
Correct me if I'm wrong but I think everything in a public interface is already publicly accessible. So I think those public modifiers are a bit redundant.